src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -298,6 +298,11 @@ private IParameterSymbol GetParameterWithMatchingName(ImmutableArray<IParameterS
                 return null;
             }
 
+            if (((IMethodSymbol)symbolInfo.Symbol).MethodKind == MethodKind.LocalFunction)
+            {
+                return null;
+            }
+
             var containingType = (INamedTypeSymbol)symbolInfo.Symbol.ContainingSymbol;
```

Comment:
i would also change this to .ContainingType.  Or at least validate that .ContainingSymbol is actually a NamedType.  Otehrwise, you'll probably just crash here in the future with some other new language construct.